### ISSUE TITLE
Use existing server instead off local jar file?

### ISSUE THREAD

**Issue by feuermandel** [antonWetzel/typst-languagetool#15]
Dear Anton,

is there any chance to use **typst-languagetool** with an existing server within my LAN (I can tunnel any port to localhost via SSH). As running the local jar file consumes a lot ram and cpu un my somehow outdated hardware.

**Comment by antonWetzel** [comment]
I added the option to use a languagetool server.
With the feature `remote-server` and the options `host=...` and `port=...` a remote server is used.
The request is only the text content without markup, so long documents should work.
The text is not seperated so really long documents might fail.

**Comment by feuermandel** [comment]
Sad to say, but I am new to rust. 

But I finally manged to compile the `cli` and the `lsb` on Windows via 

`cargo install --git=https://github.com/antonWetzel/typst-languagetool lsp --features=remote-server` and
`cargo install --git=https://github.com/antonWetzel/typst-languagetool cli --features=remote-server` 

I also managed to find the compiled binaries (`c:\users\<myuser>\.cargo\bin`)

I am able to check a typst file and a complete directory with

`typst-languagetool.exe --language "de-DE" --host http://my.server.url --port 8010 check c:\temp\typst\myfile.typ`

But when trying to use the lsp in vscode with this configuration

      "generic-lsp.configuration": [        
        {
          "lsp": "C:\\Users\\myuser\\.cargo\\bin\\typst-languagetool-lsp.exe",
          "language": "typst", 
          "options": {
            "host": "http://my.server.url",
            "port": "8010",            
            }
          },
      ]      ,

requests are made, but the lsp gets an error:

`starting LSP server
Allow Words not supported for Remote LanguageTool
unknown notification: Notification { method: "$/cancelRequest", params: Object {"id": Number(1)} }
unknown notification: Notification { method: "$/cancelRequest", params: Object {"id": Number(2)} }
unknown notification: Notification { method: "$/cancelRequest", params: Object {"id": Number(3)} }`

I am sure, I made an error.

BTW, how to I change the language to be checked from en-US to any other language?

**Comment by antonWetzel** [comment]
Changing the checked language is possible with
```json5
...
"options": {
  "language": "de-DE", // code for document language
  ...,
},
...
```

The first error happens, because there no easy way to add allowed words with a local server.
https://forum.languagetool.org/t/it-is-possible-to-add-word-with-local-server-rest-api/6479
I added post-processing step to filter the results based on the allowed words, so allowed words should work for remote now.

The unkown notifications happen, because the LSP-server gets cancel request if a file is saved again before the response. I changed the lsp to ingore them without warning.
